neurodamus/node.py

```diff
@@ -1294,6 +1294,41 @@ def _sim_corenrn_write_config(self, corenrn_restore=False):
                 self._pc.nrnbbcore_write(corenrn_data)
                 MPI.barrier()  # wait for all ranks to finish corenrn data generation
 
+        if self._shm_enabled and SimConfig.cli_options.enable_shm == "CACHE":
+            # Below improvement should only be enabled if /dev/shm is available
+            # If /dev/shm is enabled this means the the <*_{1,2,3}.dat> files are going to
+            # be written first in /dev/shm (which is going to be used as a buffer to
+            # accelerate the writes and avoid GPFS IO)
+            # Rank 0 should move the <*_{1,2,3}.dat> files to a the coreneuron_datadir on
+            # GPFS (coreneuron_input_gpfs). Each node should move them to a separate
+            # subfolder (with name NODE_ID?) of coreneuron_datadir to increase GPFS
+            # performance. We can do that by finding out the Rank 0 of every node (already
+            # implemented).
+            # We will then create links on coreneuron_datadir for each file in the
+            # subfolders.
+            local_node_rank0 = int(SHMUtil.local_ranks[0])
+            if MPI.rank == local_node_rank0:
+                import shutil
+
+                node_specific_corenrn_output_in_storage = os.path.join(
+                    corenrn_output, "coreneuron_input_" + str(SHMUtil.node_id)
```

Comment:
Maybe we could turn ` corenrn_output` to a Path object and
```
node_specific_corenrn_output = corenrn_output / "coreneuron_input_" + str(SHMUtil.node_id)
...
os.symlink(node_specific_corenrn_output / filename, f)
```

Reply:
Good idea. I improved this a bit :+1:

Reply:
Sorry for nitpicking here, but we use `os.path` everywhere in Neurodamus and there is no actual benefit to swap this to `Path`. There are also some caveats of using `Path`, so let's not add more complexity than this code already has.

Check the new version, it looks quite clean "as is". If you guys still feel this must change for some reason, then let's discuss it and we change it everywhere and not just here.